ofrak_core/ofrak/service/resource_service.py

```diff
@@ -218,21 +218,27 @@ class ResourceSortLogic(Generic[T], ABC):
     def has_effect(self) -> bool:
         raise NotImplementedError()
 
+    @abstractmethod
```

Comment:
If 1/2 implementations is allowed to not implement this, is it really an "abstract method"? It seems more like a default implementation. Perhaps just add the `#pragma: no cover` to these. What do you think?

Reply:
I considered doing it this way. I decided against it because they are not default implementations. If they were default implementations, then they should probably be tested (they are not) or invoked. 

Moreover, I don't see any other future implementation of `ResourceSortLogic` beyond `NullResourceSortLogic` using these as "default implementations" -- most of them would probably implement each of those interfaces, so they really are abstract methods that `NullResourceSortLogic` just doesn't need to implement.

Because of this, I think the above solution makes the most sense. As an alternative, I think deleting these methods would also work.

Let me know how you want me to proceed.